src/metpy/calc/tools.py

```diff
@@ -1480,3 +1480,14 @@ def _unabbrieviate_direction(abb_dir_str):
             .replace('W', 'West ')
             .replace(' ,', ',')
             ).strip()
+
+
+def _remove_nans(pressure, nan_variable):
+    """Remove NaNs from arrays that cause issues with calculations.
+
+    Takes a variable number of arguments
+    Returns a dictionary with arrays in the same order as provided
+    """
+    no_nan_variable = nan_variable[~np.isnan(nan_variable) & ~np.isnan(pressure)]
```

Comment:
I'm not sure if the docstring is out of date or not, but I *do* think it would be good to work with multiple variables at once. In fact, that might actually be necessary if you get data where dewpoint and temperature have nans in different spots. What about:
```python
def remove_nans(*variables):
    # Determine mask joint across all variables
    
    mask = None
    for v in variables:
        if mask is None:
            mask = np.isnan(v)
        else:
            mask |= np.isnan(v)

    # Mask everyone with that joint mask
    ret = []
    for v in variables:
        ret.append(v[~mask])
    return ret
```

Reply:
Thanks for this...this was what I was intending but didn't know how to do. Haven't used `|=` before - very handy here!